src/status_im/subs/community/account_selection.cljs

```diff
@@ -30,18 +30,16 @@
 
 (re-frame/reg-sub :communities/can-edit-shared-addresses?
  (fn [[_ community-id]]
-   [(re-frame/subscribe [:communities/community community-id])
-    (re-frame/subscribe [:communities/my-pending-request-to-join community-id])])
- (fn [[{:keys [joined]} pending-requests]]
-   (boolean (or joined (seq pending-requests)))))
+   [(re-frame/subscribe [:communities/community community-id])])
+ (fn [[{:keys [joined]}]]
```

Comment:
Since it's only a single args now, Do we need double vectors?

Reply:
You are correct, we indeed don't need the vector. I dislike how re-frame is inconsistent at times. Besides your suggestion, I can also remove the vector from line 33, but keep the destructuring as is and it still works...

Thanks @ibrkhalil

Reply:
I remember @flexsurfer telling me it causes memory leaks sometimes.

Reply:
I found a few other subs in `develop` (`:communities/community-color`, `:communities/community-outro-message`, `:communities/community-members`, etc) that have the extra/unnecessary vector in the destructuring part, but I don't understand how that could cause a leak.

When there's only one input signal, re-frame supports the first argument from the sub handler to be either a vector or a single argument. If we truly identified a leak due to this API detail, then we should definitely report it as a bug.

Reply:
Oh @ibrkhalil, I think I remember now what you mean. Sometimes a developer can forget to remove an input signal, even thought it's not used anymore in the handler. In that way, this causes re-frame to do more work than necessary. For example (artificial, not in `develop`):

```clojure
(re-frame/reg-sub :communities/community-joined
 (fn [[_ community-id]]
   [(re-frame/subscribe [:communities/community community-id])
    (re-frame/subscribe [:communities/community-color community-id])])
 (fn [[{:keys [joined]}] _]
   joined))
```

See ya :wave: 🤝